utils/server/websocket.go

```diff
@@ -247,58 +267,75 @@ func (s *Server) handleGraphqlSocket(adminMan *admin.Manager) http.HandlerFunc {
 
 				for _, dirValue := range v.Arguments {
-					if dirValue.Name.Value == "where" {
-						whereData, err := graphql.ExtractWhereClause(v.Arguments, utils.M{})
-						if err != nil {
-							log.Println(err)
-							return
-						}
-						data.Where = whereData
-					} else if dirValue.Name.Value == "skipInitial" {
+					if dirValue.Name.Value == "skipInitial" {
 						if dirValue.Value.(*ast.BooleanValue).Value {
 							data.Options = model.LiveQueryOptions{SkipInitial: true}
 						}
 					}
 				}
+
+				whereData, err := graphql.ExtractWhereClause(v.Arguments, utils.M{})
+				if err != nil {
+					channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GQL_ERROR, Error: err.Error()}})
+					closeConnAliveRoutine <- true
+					return
+				}
+				data.Where = whereData
 				data.Token = token
-				data.DBType = v.Directives[0].Name.Value
+				data.DBType = graphql.GetDBType(v)
 				data.Project = project
 				data.Group = v.Name.Value
 				data.Type = m.Type
 				data.ID = m.ID
 				graphqlIDMapper.Store(m.ID, data.Group)
 
 				// Subscribe to realtime feed
-				feedData, err := s.realtime.Subscribe(ctx, m.ID, s.auth, s.crud, data, func(feed *model.FeedData) {
-					channel <- (&graphqlMessage{ID: feed.QueryID, Payload: payloadObject{Status: utils.GqlData, Data: graphqlSucessData{Type: feed.Type, Doc: feed.Payload, DocID: feed.DocID}}})
+				feedData, err := s.realtime.Subscribe(ctx, clientID, s.auth, s.crud, data, func(feed *model.FeedData) {
+					channel <- (&graphqlMessage{ID: feed.QueryID, Payload: payloadObject{Status: utils.GQL_DATA, Data: graphqlSucessData{Type: feed.Type, Doc: feed.Payload, DocID: feed.DocID}}})
 				})
 
 				if err != nil {
-					channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GqlError, Error: err.Error()}})
+					channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GQL_ERROR, Error: err.Error()}})
+					closeConnAliveRoutine <- true
 					return
 				}
 
-				channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GqlStart, Data: feedData}})
+				channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GQL_START, Data: feedData}})
 
-			case utils.GqlStop:
-				// For realtime subscribe event
+			case utils.GQL_STOP:
 				group, ok := graphqlIDMapper.Load(m.ID)
 				if !ok {
+					channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GQL_ERROR, Error: errors.New("Got " + utils.GQL_STOP + " wanted " + utils.GQL_START).Error()}})
+					closeConnAliveRoutine <- true
 					return
 				}
 				data := new(model.RealtimeRequest)
 				data.ID = m.ID
 				data.Group = group.(string)
 
-				s.realtime.Unsubscribe(m.ID, data)
-				channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GqlStop}})
-
+				s.realtime.Unsubscribe(clientID, data)
+				channel <- (&graphqlMessage{ID: m.ID, Payload: payloadObject{Status: utils.GQL_STOP}})
+				graphqlIDMapper.Delete(m.ID)
```

Comment:
should I delete the id after unsubscribe

Reply:
Yes